lib/ansible/modules/files/jsonpath.py

```diff
@@ -0,0 +1,206 @@
+    type: str
+    required: yes
+
+  value:
+    description:
+    - Either a string, or number.
+    - This parameter is required.
+    type: raw
+    required: yes
+
+requirements:
+- jsonpath-ng >= 1.4.3
+notes:
+- This module does not handle complicated jsonpath expressions, so limit jsonpath selectors to simple expressions.
+
+author:
+- zhan9san (@zhan9san)
+'''
+
+EXAMPLES = r'''
+# Consider the following JSON file:
+#
+# {
+#   "menu": {
+#     "id": "file",
+#     "value": "File",
+#     "popup": {
+#       "menuitem": [{
+#           "value": "New",
+#           "onclick": "CreateNewDoc()"
+#         },
+#         {
+#           "value": "Open",
+#           "onclick": "OpenDoc()"
+#         },
+#         {
+#           "value": "Close",
+#           "onclick": "CloseDoc()"
+#         }
+#       ]
+#     }
+#   }
+# }
+
+- name: Set menuitem's id to '01'
+  jsonpath:
+    path: /foo/bar.json
+    jsonpath: "menu['id']"
+    value: '01'
+
+- name: Set the first menuitem's onclick to 'NewDoc()'
+  jsonpath:
+    path: /foo/bar.json
+    jsonpath: "menu['popup']['menuitem'][0]['onclick']"
+    value: 'NewDoc()'
+'''
+
+RETURN = r'''
+actions:
+    description: A dictionary with the original jsonpath.
+    type: dict
+    returned: success
+    sample: {jsonpath: jsonpath}
+
+msg:
+    description: A message related to the performed action(s).
+    type: str
+    returned: always
+'''
+
+import copy
+import json
+import os
+
+from distutils.version import LooseVersion
+
+try:
+    from jsonpath_ng import parse
+    from jsonpath_ng import __version__ as jsonpath_ng_version
+    HAS_JSONPATH_NG = True
+except ImportError:
+    HAS_JSONPATH_NG = False
+
+from ansible.module_utils.basic import AnsibleModule, json_dict_bytes_to_unicode
+from ansible.module_utils._text import to_native
+
+
+def set_target(module, data, jsonpath_expr, orig_value):
+    if orig_value == module.params['value']:
+        changed = False
+    else:
+        jsonpath_expr.update(data, module.params['value'])
+        changed = True
+
+    finish(module, data, changed)
+
+
+def finish(module, data, changed=False, msg=''):
+
+    result = dict(
+        actions=dict(
+            jsonpath=module.params['jsonpath'],
+        ),
+        changed=changed,
+    )
+
+    if msg:
+        result['msg'] = msg
+
+    if result['changed']:
+        with open(module.params['path'], 'w') as f:
+            try:
+                json.dump(data, f, ensure_ascii=False, indent=2, sort_keys=True)
+            except Exception as e:
+                module.fail_json(msg="Write error in json file: %s (%s)" % (module.params['path'], e))
+
+    module.exit_json(**result)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            path=dict(type='path', required=True),
+            jsonpath=dict(type='str', required=True),
+            value=dict(type='raw', required=True),
+        ),
+        supports_check_mode=False,
```

Comment:
This module should support `check_mode`.

Reply:
@ptux 
Thanks for your suggestion. I have enabled check_mode.
Would you please review it again?